avogadro/rendering/solid_first_fs.glsl

```diff
@@ -136,18 +184,25 @@ float computeEdgeLuminosity(vec3 normal)
 }
 
 void main() {
+  // Some cleanups required.
   float luminosity = 1.0;
   luminosity *= max(1.2 * (1.0 - inAoEnabled), computeSSAOLuminosity(getNormalNear(UV)));
   luminosity *= max(1.0 - inEdStrength, computeEdgeLuminosity(getNormalAt(UV)));
   vec4 color = texture2D(inRGBTex, UV);
-  if (inFogStrength == 0.0) {
+  if (inFogStrength == 0.0 && inDofEnabled == 0.0) {
     gl_FragColor = vec4(color.xyz * luminosity, color.w);
   }
-  else {
+  else if (inFogStrength != 0.0) { // apply fog since it's fast.
```

Comment:
I think we may want a separate shader path with fog, but not AO or depth blur (e.g., for speed, less-powerful GPU, etc.). I'll follow up on the forum.